### PR TITLE
Fix generating join aliases for subcriteria

### LINKED ISSUE #2175: Subcriteria on component collection generates incorrect join alias
After upgrading from NHibernate 5.1 to 5.2 a unit tests failed due to incorrectly generated alias for subcriteria on a component collection: 

    SQLite Error 1: 'no such column: code2_.code_source_id'.

The generated SQL looks as follows:

    SELECT this_.concept_id as concept1_3_2_
        , conceptcod5_.code_id as code1_1_0_
        , conceptcod5_.code_source_id as code4_1_0_
        , conceptcod5_.value as value5_1_0_
        , conceptcod5_.name as name6_1_0_
    FROM concept this_ 
    inner join concept_mapping cm1_ on this_.concept_id=cm1_.concept_id 
    left outer join concept_code conceptcod5_ on cm1_.code_id=conceptcod5_.code_id 
    WHERE code2_.code_source_id in (@p0) 
    and code2_.value in (@p1, @p2, @p3)

The problem is that in the WHERE clause the SQL alias `code2_` is used for table `concept_code` but in the FROM clause the SQL alias `conceptcod5_` is defined for the same table.

### PR DESCRIPTION
**Pull request by ggeurts**
Fixes #2175

### CODE REVIEW

File: src/NHibernate.Test/NHibernate.Test.csproj [hunk 1]
```diff
@@ -88,4 +88,8 @@
   <ItemGroup>
     <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
   </ItemGroup>
+  <ItemGroup>
+    <Folder Include="NHSpecificTest\NH2875\" />
```
Comment (hazzik) [inline src/NHibernate.Test/NHibernate.Test.csproj]: Shouldn’t be here

**Comment by ggeurts** [comment]
Work on this PR is complete

**Comment by bahusoid** [comment]
Async counterpart is missing for tests. And it seems something is wrong with your test on Firebird

**Comment by ggeurts** [comment]
Many thanks for the fixes

**Pull request merged**